pkg/daemon/ceph/osd/kms/volumes.go

```diff
@@ -82,25 +82,32 @@ func VaultVolumeAndMount(config map[string]string) (v1.Volume, v1.VolumeMount) {
 func tlsSecretPath(tlsOption string) string {
 	switch tlsOption {
 	case api.EnvVaultCACert:
-		return vaultCAFileName
+		return VaultCAFileName
 	case api.EnvVaultClientCert:
-		return vaultCertFileName
+		return VaultCertFileName
 	case api.EnvVaultClientKey:
-		return vaultKeyFileName
+		return VaultKeyFileName
 
 	}
 
 	return ""
 }
 
 // VaultTokenFileVolume save token from secret as volume mount
-func VaultTokenFileVolume(tokenSecretName string) v1.Volume {
-	return v1.Volume{
-		Name: secrets.TypeVault,
+func VaultTokenFileVolume(tokenSecretName string) (v1.Volume, v1.VolumeMount) {
+	volume := v1.Volume{
+		Name: "vault-file-vol",
```

Comment:
Why not keeping `secrets.TypeVault`?

Reply:
@leseb : If TLS enabled then it will end up having two volumes having the same name, one with TLS certs for vault and the other with the token file. I didn't modify `VaultVolumeAndMount()` to include the `token` file since it is not required for the OSD encryption and wrote separate function for that